homeassistant/components/freebox/__init__.py

```diff
@@ -37,54 +35,70 @@
-        port = conf.get(CONF_PORT)
-        await async_setup_freebox(hass, config, host, port)
+    if conf is None:
+        return True
+
+    hass.async_create_task(
+        hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_IMPORT}, data=conf,
+        )
+    )
 
     return True
 
 
-async def async_setup_freebox(hass, config, host, port):
-    """Start up the Freebox component platforms."""
+async def async_setup_entry(hass: HomeAssistantType, entry: ConfigEntry):
+    """Set up Freebox component."""
+    router = FreeboxRouter(hass, entry)
+    await router.setup()
 
-    app_desc = {
-        "app_id": "hass",
-        "app_name": "Home Assistant",
-        "app_version": "0.65",
-        "device_name": socket.gethostname(),
-    }
+    hass.data.setdefault(DOMAIN, {})
+    hass.data[DOMAIN][entry.unique_id] = router
 
-    token_file = hass.config.path(FREEBOX_CONFIG_FILE)
-    api_version = "v6"
+    for platform in PLATFORMS:
+        hass.async_create_task(
+            hass.config_entries.async_forward_entry_setup(entry, platform)
+        )
 
-    fbx = Freepybox(app_desc=app_desc, token_file=token_file, api_version=api_version)
+    # Services
+    async def async_reboot(call):
+        """Handle reboot service call."""
+        await router.reboot()
 
-    try:
-        await fbx.open(host, port)
-    except HttpRequestError:
-        _LOGGER.exception("Failed to connect to Freebox")
-    else:
-        hass.data[DATA_FREEBOX] = fbx
+    hass.services.async_register(DOMAIN, "reboot", async_reboot)
 
-        async def async_freebox_reboot(call):
-            """Handle reboot service call."""
-            await fbx.system.reboot()
+    async def async_close_connection(event):
+        """Close Freebox connection on HA Stop."""
+        await router.close()
 
-        hass.services.async_register(DOMAIN, "reboot", async_freebox_reboot)
+    hass.bus.async_listen_once(EVENT_HOMEASSISTANT_STOP, async_close_connection)
 
-        hass.async_create_task(async_load_platform(hass, "sensor", DOMAIN, {}, config))
-        hass.async_create_task(
-            async_load_platform(hass, "device_tracker", DOMAIN, {}, config)
-        )
-        hass.async_create_task(async_load_platform(hass, "switch", DOMAIN, {}, config))
+    return True
 
-        async def close_fbx(event):
-            """Close Freebox connection on HA Stop."""
-            await fbx.close()
 
-        hass.bus.async_listen_once(EVENT_HOMEASSISTANT_STOP, close_fbx)
+async def async_unload_entry(hass: HomeAssistantType, entry: ConfigEntry):
+    """Unload a config entry."""
+    unload_ok = all(
+        await asyncio.gather(
+            *[
+                hass.config_entries.async_forward_entry_unload(entry, platform)
+                for platform in PLATFORMS
+            ]
+        )
+    )
+    if unload_ok:
+        router = hass.data[DOMAIN][entry.unique_id]
+        await router.close()
+        hass.data.pop(DOMAIN)
```

Comment:
If there are more than one entry we can't pop the domain here. Either add a check or just leave the domain key in hass.data.

Reply:
Oooops